src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -2523,6 +2530,63 @@ class CreateStatementSegment(BaseSegment):
         ),
         Ref("IfNotExistsGrammar", optional=True),
         Ref("ObjectReferenceSegment"),
+        # Next set are Storage Integration statements
+        # https://docs.snowflake.com/en/sql-reference/sql/create-storage-integration.html
+        Sequence(
+            AnySetOf(
+                Sequence("TYPE", Ref("EqualsSegment"), "EXTERNAL_STAGE"),
+                Sequence("ENABLED", Ref("EqualsSegment"), Ref("BooleanLiteralGrammar")),
+                OneOf(
+                    Ref("S3StorageIntegrationParameters"),
+                    Ref("GCSStorageIntegrationParameters"),
+                    Ref("AzureStorageIntegrationParameters"),
+                ),
+                Sequence(
+                    "STORAGE_ALLOWED_LOCATIONS",
+                    Ref("EqualsSegment"),
+                    OneOf(
+                        Bracketed(
+                            Delimited(
+                                OneOf(
+                                    Ref("S3Path"),
+                                    Ref("GCSPath"),
+                                    Ref("AzureBlobStoragePath"),
+                                )
+                            )
+                        ),
+                        Bracketed(
+                            Ref("QuotedStarSegment"),
+                        ),
+                    ),
+                ),
+                Sequence(
+                    "STORAGE_BLOCKED_LOCATIONS",
+                    Ref("EqualsSegment"),
+                    OneOf(
+                        Bracketed(
+                            Delimited(
+                                OneOf(
+                                    Ref("S3Path"),
+                                    Ref("GCSPath"),
+                                    Ref("AzureBlobStoragePath"),
+                                )
+                            )
+                        ),
+                        Bracketed(
+                            Ref("QuotedStarSegment"),
+                        ),
+                    ),
+                    optional=True,
+                ),
```

Comment:
As these are very similar I wonder if they can be combined? As they are part of an `AnySetOf` it should still be allowed to have both.

Also the second one ahas an `optional=True` on line 2579 but don't think that's needed as it's part of an `AnySetOf` so optionality is implied.

```suggestion
                Sequence(
                    OneOf("STORAGE_ALLOWED_LOCATIONS", "STORAGE_BLOCKED_LOCATIONS"),
                    Ref("EqualsSegment"),
                    OneOf(
                        Bracketed(
                            Delimited(
                                OneOf(
                                    Ref("S3Path"),
                                    Ref("GCSPath"),
                                    Ref("AzureBlobStoragePath"),
                                )
                            )
                        ),
                        Bracketed(
                            Ref("QuotedStarSegment"),
                        ),
                    ),
                ),
```